lib/routes.js

```diff
@@ -279,14 +283,20 @@ module.exports = async function (app, opts) {
     schema: postSchema(allowBatchedQueries),
     attachValidation: true
   }, async function (request, reply) {
+    // Generate the context for this request
+    const context = contextFn ? await contextFn(request, reply) : {}
+
+    request[kRequestContext] = context
+    Object.assign(request[kRequestContext], { reply, app })
```

Comment:
If `contextFn` is not set, this `Object.assign()` could be avoided.

Reply:
[defaultErrorFormatter](https://github.com/wiktor-obrebski/mercurius/blob/polishing/error-formatter-context/lib/errors.js#L64) assumes that if there is any context (even empty object) there is `reply` or `app` property available on it.

before it was not a problem, as the `errorFormatter` just did not have the context available in some cases. but now it have it always.